### PR TITLE
Feat/select recipes in header plugin

### LINKED ISSUE #77: feat: update the header plugin to be able to select what ui recipe to display below the header
The icon on the left should open a list of possible ui recipes to select. 

Image

### PR DESCRIPTION
**Pull request by KristianKjerstad**
## What does this pull request change?

Add functionality to select a UI plugin in the header
image

Also, refactor the Header.tsx file - split up into components for `AboutDialog` and `UserInfoDialog` 

## Issues related to this change
closes #77

### CODE REVIEW

File: packages/header/src/Header.tsx [hunk 1]
```diff
@@ -27,208 +27,105 @@ const Icons = styled.div`
...
 export default (props: IUIPlugin): JSX.Element => {
   const { idReference, config: passedConfig } = props
-  const config: THeaderPluginConfig = passedConfig
-  const [entity, isLoading] = useDocument<TApplication>(idReference)
-  const { tokenData, token, logOut } = useContext(AuthContext)
+  const config = passedConfig as THeaderPluginConfig
+  const [entity, isApplicationLoading] = useDocument<TApplication>(idReference)
+
+  const path = 'dmss://DemoDataSource/DemoPackage/ExampleApplication'
+  const idRef = 'DemoDataSource/46a2c6ce-5bb4-41bc-923e-ede8dc95500e'
```
Comment (KristianKjerstad) [inline packages/header/src/Header.tsx:49]: Not sure how to deal with these two hard coded values. Perhaps let is be part of config instead?

File: packages/header/src/components/UserInfoDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import { Icon, Radio, TopBar } from '@equinor/eds-core-react'
```
Comment (KristianKjerstad) [inline packages/header/src/components/UserInfoDialog.tsx:1]: this stuff is copied from old Header plugin

File: example/app/data/DemoDataSource/DemoPackage/recipe_links/example_application.json [hunk 3]
```diff
@@ -4,7 +4,13 @@
   "initialUiRecipe": {
     "name": "Header",
     "type": "CORE:UiRecipe",
-    "plugin": "header"
+    "plugin": "header",
+    "config": {
+      "uiPluginsList": ["yaml-view", "explorer"],
```
Comment (soofstad) [inline example/app/data/DemoDataSource/DemoPackage/recipe_links/example_application.json:9]: I think you should select recipes here, not plugins.

File: packages/header/README.md [hunk 4]
```diff
@@ -1,10 +1,21 @@
...
 If you want to specify a config object in some UiRecipe-entity, the Blueprint is included in the npm package.
 
 Copy it like so: `cp -R node_modules/@development-framework/header/dist/blueprint ./myApplicationBlueprints/`
 
+# Config
+
+The Header plugin can accept a config of type `HeaderPluginConfig` (see `blueprints/HeaderPluginConfig.json`).
+Explanation of the attributes:
+
+* `uiPluginsList`: Can be used to specify a list of UI plugins the user can select. By default, the fist UI plugin in
```
Comment (soofstad) [inline packages/header/README.md:17]: Recipes, not plugins

File: packages/header/README.md [hunk 5]
```diff
@@ -1,10 +1,21 @@
...
+  the list is displayed. If the list is empty, no plugin is displayed. NB! All plugins listed here must be included in
```
Comment (soofstad) [inline packages/header/README.md:18]: Really think the header should use all recipes if the "uiPluginsList" is empty.
Sane defaults and minimum amount of config being the goal.

File: packages/header/package.json [hunk 6]
```diff
@@ -1,9 +1,17 @@
...
-    "@development-framework/dm-core": ">=1.0.35",
+    "@development-framework/dm-core": ">=1.0.38",
+    "@development-framework/yaml-view": ">=1.0.4",
+    "@development-framework/explorer": ">=1.0.0",
+    "@development-framework/default-pdf": "^1.0.5",
+    "@development-framework/default-preview": "^1.0.2",
+    "@development-framework/blueprint": "^1.0.5",
+    "@development-framework/mermaid": "^1.0.2",
+    "@development-framework/tabs": "^1.0.3",
+    "@development-framework/form": "^1.0.0",
```
Comment (soofstad) [inline packages/header/package.json:14]: Should not be added here.
Instead should be loaded from the dm-core "UiPluginsContext". Similar to what is done in "UiPluginsSelector".

File: packages/header/src/components/PluginSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  &:hover {
+    background: #4f878d;
+  }
+`
+
+type RecipeSelectorProps = {
+  uiPluginNames: string[]
+  setSelectedUiPlugin: (uiPlugin: string) => void
+}
+export const PluginSelector = (props: RecipeSelectorProps) => {
```
Comment (soofstad) [inline packages/header/src/components/PluginSelector.tsx:41]: Make sure not to use "Plugin" and "Recipe" interchangeably. They are not the same :)

**Review by soofstad — CHANGES_REQUESTED** [review]

File: example/app/data/DemoDataSource/DemoPackage/HeaderPluginConfig.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      "attributeType": "boolean",
+      "type": "dmss://system/SIMOS/BlueprintAttribute",
+      "name": "hideUserInfo",
+      "default": false
+    },
+    {
+      "attributeType": "string",
+      "type": "dmss://system/SIMOS/BlueprintAttribute",
+      "dimensions": "*",
+      "name": "uiPluginsList"
```
Comment (soofstad) [inline example/app/data/DemoDataSource/DemoPackage/HeaderPluginConfig.json:21]:
```suggestion
      "name": "uiRecipesList"
```

File: packages/header/src/Header.tsx [hunk 9]
```diff
@@ -27,208 +27,113 @@ const Icons = styled.div`
...
+          {appSelectorOpen && (
+            <PluginSelector
+              selectableUiRecipeNames={
+                config.uiRecipesList.length > 0
+                  ? config.uiRecipesList
+                  : uiRecipes.map((recipe: TUiRecipe) => recipe.name)
+              }
+              availableUiRecipes={uiRecipes}
+              setSelectedUiPlugin={(uiPluginName) => {
+                setSelectedPlugin({ component: getUiPlugin(uiPluginName) })
```
Comment (soofstad) [inline packages/header/src/Header.tsx:102]:
```suggestion
                setAppSelectorOpen(false)
                setSelectedPlugin({ component: getUiPlugin(uiPluginName) })
```

File: packages/header/src/plugins.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+export default [
+  import('@development-framework/yaml-view'),
+  import('@development-framework/explorer'),
+  import('@development-framework/default-pdf'),
```
Comment (soofstad) [inline packages/header/src/plugins.js:4]: This file should be deleted no?

File: packages/header/src/types.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
+export type TApplication = {
+  name: string
+  type: string
+  dataSources: string[]
+  label: string
+  description: string
+  roles: string[]
+  adminRole: string
```
Comment (soofstad) [inline packages/header/src/types.ts:8]: These should be optional. 
Not every app will need them, but they should still be able to use the header plugin

**Review by soofstad — APPROVED** [review]

**Pull request merged**